src/event.cc

```diff
@@ -1,325 +1,251 @@
     } else {
-      rc = oboe_metadata_tostr(md, buf, sizeof(buf) - 1);
+      status = oboe_event_add_info(event, key.c_str(), s.c_str());
     }
+  }
 
-    return Napi::String::New(info.Env(), rc == 0 ? buf : "");
+  return status;
 }
 
 //
-// JavaScript callable method to set the event's sample flag to the boolean
-// sense of the argument.
+// add an edge
 //
-// returns the previous value of the flag.
-//
-Napi::Value Event::setSampleFlagTo(const Napi::CallbackInfo& info) {
-    Napi::Env env = info.Env();
+int addEdge(oboe_event_t* event, Napi::Value edge) {
+  // oboe returns -1 for errors, so this returns an error of 1
+  // so the two causes can be distinguished.
+  int status = 1;
 
-    if (info.Length() != 1) {
-        Napi::Error::New(env, "setSampleFlagTo requires one argument").ThrowAsJavaScriptException();
-        return env.Null();
-    }
+  if (!edge.IsObject() || edge.IsArray()) {
+    return status;
+  }
 
-    bool previous = this->event.metadata.flags & XTR_FLAGS_SAMPLED;
+  Napi::Object mb = edge.As<Napi::Object>().Get("buf").As<Napi::Object>();
+  Napi::Buffer<uint8_t> buf = mb.As<Napi::Buffer<uint8_t>>();
 
-    // truthy to set it, falsey to clear it
-    if (info[0].ToBoolean().Value()) {
-        this->event.metadata.flags |= XTR_FLAGS_SAMPLED;
-    } else {
-        this->event.metadata.flags &= ~XTR_FLAGS_SAMPLED;
-    }
-    return Napi::Boolean::New(env, previous);
-}
+  oboe_metadata_t oboe_md;
+  initializeOboeMd(oboe_md, buf);
 
-//
-// JavaScript callable method to get the sample flag from
-// the event metadata.
-//
-// returns boolean
-//
-Napi::Value Event::getSampleFlag(const Napi::CallbackInfo& info) {
-    return Napi::Boolean::New(info.Env(), this->event.metadata.flags & XTR_FLAGS_SAMPLED);
+  status = oboe_event_add_edge(event, &oboe_md);
+
+  return status;
 }
 
 //
-// JavaScript callable method to get a new metadata object with
-// the same metadata as this event.
+// Event send function. The edges and KVs have been validated
+// by the JavaScript Event class so minimal checking is required.
 //
-// Event.getMetadata()
-//
-// Return a Metadata object with the metadata from this event.
-//
-Napi::Value Event::getMetadata(const Napi::CallbackInfo& info) {
-    Napi::Env env = info.Env();
+Napi::Value send(const Napi::CallbackInfo& info) {
+  Napi::Env env = info.Env();
 
-    oboe_metadata_t* omd = &this->event.metadata;
+  if (info.Length() < 1 || !info[0].IsObject()) {
+    Napi::TypeError::New(env, "Invalid signature").ThrowAsJavaScriptException();
+    return env.Null();
+  }
 
-    Napi::Value v = Napi::External<oboe_metadata_t>::New(env, omd);
-    return Metadata::NewInstance(env, v);
-}
+  // the default is to use the OBOE_SEND_EVENT channel. A truthy second argument
+  // will result in using the OBOE_SEND_STATUS channel.
+  Napi::Object event = info[0].As<Napi::Object>();
+  const int channel = info[1].ToBoolean().Value() ? OBOE_SEND_STATUS : OBOE_SEND_EVENT;
```

Comment:
good question. no, it can't. accessing a non-existent index in `info` results in an undefined value (which mirrors javascript behavior so is kind of nice). and `undefined` converted to a `boolean` is false, so it works as expected.

i should remove the commented out //bool useStatusChannel though.